### ISSUE TITLE
Array of pointer not diagnosed

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#34779]
As observed by https://github.com/dotnet/roslyn/issues/34693#issuecomment-479675820 the following program fails to report the incorrect use of a pointer type after https://github.com/dotnet/roslyn/pull/34739

``` c#
public class A {
    public class B {}
}

public class C<T> : A {
    public C<string*[]>.B field; // error
}
```

**Comment by RikkiGibson** [comment]
I wasn't precise enough in my comment there. As far as I can tell, the first case with `string*` is diagnosed because it's checked at the time `C<string*[]>` is bound--and we [have a test for this](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests/Semantics/UnsafeTests.cs,8280). The second case however isn't caught because we check the constraints after fields are bound.

As @cston pointed out there is a similar bug with method return types in 2.9.0 and today:

```cs
public class A
{
    public class Inner {}
}

public class B<T> : A where T : unmanaged
{
}

public class C
{
    B<string>.Inner M() => throw null;
}
```

This bug is likely to be present in every case where constraints are checked after binding instead of during binding.

**Comment by gafter** [comment]
In that case, moving to Backlog.
